pins/databricks/fs.py

```diff
@@ -0,0 +1,186 @@
+    if file_type == "file":
+        board.fs.get(rpath, lpath, **kwargs)
+        return
+
+    def _get_files(path, recursive, **kwargs):
+        raw_contents = w.files.list_directory_contents(path)
+        contents = list(raw_contents)
+        details = list(map(_databricks_content_details, contents))
+        for item in details:
+            item_path = item.get("path")
+            if item.get("is_directory"):
+                if recursive:
+                    _get_files(item_path, recursive=recursive, **kwargs)
+            else:
+                rel_path = os.path.relpath(item_path, rpath)
+                target_path = os.path.join(lpath, rel_path)
+                board.fs.get(item_path, target_path)
+
+    _get_files(rpath, recursive, **kwargs)
+
+
+def _databricks_open(path):
+    if not _databricks_exists(path):
+        raise PinsError("File or directory does not exist")
+    w = WorkspaceClient()
+    resp = w.files.download(path)
+    f = BytesIO()
+    shutil.copyfileobj(resp.contents, f)
+    f.seek(0)
+    return f
+
+
+def _databricks_exists(path: str):
+    if _databricks_is_type(path) == "nothing":
+        return False
+    else:
+        return True
+
+
+def _databricks_is_type(path: str):
+    w = WorkspaceClient()
+    try:
+        w.files.get_metadata(path)
+    except NotFound:
+        try:
+            w.files.get_directory_metadata(path)
+        except NotFound:
+            return "nothing"
+        else:
+            return "directory"
+    else:
+        return "file"
+
+
+def _databricks_ls(path, detail):
+    if not _databricks_exists(path):
+        raise PinsError("File or directory does not exist")
+    w = WorkspaceClient()
+    if _databricks_is_type(path) == "file":
+        if detail:
+            return [dict(name=path, size=None, type="file")]
+        else:
+            return path
+
+    contents_raw = w.files.list_directory_contents(path)
+    contents = list(contents_raw)
+    items = []
+    for item in contents:
+        item = _databricks_content_details(item)
+        item_path = item.get("path")
+        item_path = item_path.rstrip("/")
+        if detail:
+            if item.get("is_directory"):
+                item_type = "directory"
+            else:
+                item_type = "file"
+            items.append(dict(name=item_path, size=None, type=item_type))
+        else:
+            items.append(item_path)
+    return items
+
+
+def _databricks_rm_dir(path):
+    w = WorkspaceClient()
+    raw_contents = w.files.list_directory_contents(path)
+    contents = list(raw_contents)
+    details = list(map(_databricks_content_details, contents))
+    for item in details:
+        item_path = item.get("path")
+        if item.get("is_directory"):
+            _databricks_rm_dir(item_path)
+        else:
+            w.files.delete(item_path)
+    w.files.delete_directory(path)
+
+
+def _databricks_mkdir(path):
+    w = WorkspaceClient()
+    w.files.create_directory(path)
```

Comment:
What does this do if the directory already exists?

Reply:
If the directory exists it does nothing, not even a warning